stocks/stocks.py

```diff
@@ -1,7 +1,155 @@
+    """
+    Make sure symbol is trackable
+    """
+    r = requests.get(STOCK_PROFILE_URL.format(token=FINNHUB_TOKEN, symbol=symbol))
+    response = r.json()
+    if response: 
+        return response.get("name")
+    return False
+
+
+
+def calculate_percent_change(response, average_cost):
+    """
+    Calculate percent change in stock compared to average cost
+    """
+    current_price = response["c"]
+    difference = current_price - average_cost
+    percent_difference = round(difference / average_cost * 100, 2)
+    return percent_difference
+    
+
+def insert_stock_tracker(stock_id, average_cost, percent, increase, decrease):
+    """
+    Insert tracked stock average cost, percent, increase, decrease into database
+    """
+    con = sql.connect(config.DATABASE) 
+    cur = con.cursor()
+    
+    cur.execute("SELECT id FROM stock_tracker WHERE stock_id=?", [stock_id]) 
+    if cur.fetchone():
+        cur.execute("DELETE FROM stock_tracker WHERE stock_id=?", [stock_id])
+        con.commit()
+    cur.execute("INSERT INTO stock_tracker (average_cost, percent, increase, decrease, stock_id) VALUES(?,?,?,?,?)", (average_cost, percent, increase, decrease, stock_id))
+    con.commit()
+
+
+def filter_tracked_stocks_details(symbol, detailed):
+    con = sql.connect(config.DATABASE) 
+    cur = con.cursor()
+        
+    if symbol:
+        cur.execute("SELECT * FROM stock_tracker WHERE stock_id = (SELECT id from stock WHERE symbol=?)", [symbol]) 
+        tracked_stocks = [cur.fetchone()]
+    else:
+        cur.execute("SELECT * FROM stock_tracker")
+        tracked_stocks = cur.fetchall()
+
+    if any(stock is None for stock in tracked_stocks):
+        return 
+
+    tracked_stocks_list = []
+    for stock_details in tracked_stocks:
+        cur.execute("SELECT symbol, name FROM stock WHERE id=?", [stock_details[6]]) 
+        stock_profile = cur.fetchone()
+        symbol = stock_profile[0]
+        name = stock_profile[1]
+        average_cost = stock_details[1]
+        percent = stock_details[2]
+        increase = True if stock_details[3] else False
+        decrease = True if stock_details[4] else False
+        last_modified = stock_details[5]
+
+        if detailed:
+            response = get_stock_quote(symbol)
+            if not response:
+                return jsonify({"Error": "You are tracking an invalid stock symbol: {symbol}".format(symbol=symbol)}), 404        
+            percent_difference = calculate_percent_change(response, average_cost)
+            tracked_stocks_list.append({"symbol": symbol, "name": name, "percent_difference": percent_difference,
+                                        "last_modified": last_modified, "alert_on_increase": increase, "alert_on_decrease": decrease, 
+                                        "average_cost": average_cost, "percent_to_track_threshold": percent})
+        else:
+            tracked_stocks_list.append({"symbol": symbol, "name": name})
```

Comment:
nit: The standard response is a subset of the detailed response, so you could move the standard response creation up so that it always does that, then if detailed == True, add detailed items

resp = standard_resp

if detailed == True:
    # add fields
```